runtime/src/accounts_db.rs

```diff
@@ -6695,6 +6702,7 @@ impl AccountsDb {
                         &self.account_indexes,
                     );
                 }
+                accounts_data_len += stored_account.data().len() as u64;
```

Comment:
I decided on an implementation for `generate_index_for_slot()` that does _not_ pass in the Atomic, but instead computes the len locally and returns it at the end (as a regular integer). I figured this would be faster due to less contention on the Atomic (and avoiding the atomic operations entirely in this function).